TLM/TLM/Util/Record/SpeedLimitLaneRecord.cs

```diff
@@ -43,29 +42,35 @@ public class SpeedLimitLaneRecord : IRecordable {
                 action: SetSpeedLimitAction.FromNullableFloat(this.speedLimit_));
         }
 
+        /// <summary>
+        /// Obtain speed limit records for applicable lanes within the segment.
+        /// </summary>
+        /// <param name="segmentId">The id of the segment to inspect.</param>
+        /// <returns>
+        /// Returns a list of <see cref="SpeedLimitLaneRecord"/> for all
+        /// speed-applicable lanes in the segment. The list may be empty if no
+        /// matching lanes were found.
+        /// </returns>
         public static List<SpeedLimitLaneRecord> GetLanes(ushort segmentId) {
-            int maxLaneCount = segmentId.ToSegment().Info.m_lanes.Length;
-            var ret = new List<SpeedLimitLaneRecord>(maxLaneCount);
-            ExtSegmentManager extSegmentManager = ExtSegmentManager.Instance;
-            var lanes = extSegmentManager.GetSortedLanes(
-                segmentId,
-                ref segmentId.ToSegment(),
+
+            var lanes = segmentId.ToSegment().GetSortedLanes(
                 null,
                 LANE_TYPES,
                 VEHICLE_TYPES,
                 sort: false);
+
+            var ret = new List<SpeedLimitLaneRecord>(lanes.Count);
```

Comment:
Same comment here, can this be zero size? If so, return a premade zero size static readonly list.

Reply:
> If so, return a premade zero size static readonly list.

That would be `Enumerable.Empty<SpeedLimitRecord>()`

Reply:
wait that does not return a list